Wolverine.Weather.Infrastructure/Repositories/WeatherForecastRepository.cs

```diff
@@ -56,15 +56,15 @@ public async Task<IEnumerable<WeatherForecast>> GetWeatherForecasts(Cancellation
             }
         }
 
-        public async Task<WeatherForecast> GetWeatherForecast(int id, CancellationToken cancellationToken)
+        public async Task<WeatherForecast> GetWeatherForecast(Guid ExternalId, CancellationToken cancellationToken)
         {
             try
             {
                 string storedProcedure = _databaseConfiguration.StoredProcedureNames.WeatherForecastGetByIdStoredProcedureName;
 
                 var parameters = new DynamicParameters();
 
-                parameters.Add("@Id", id, DbType.Int32);
+                parameters.Add("@pExternalId", ExternalId, DbType.Guid);
```

Comment:
This will force you to update the name of the parameter in the database too, FYI

Reply:
You're right, I updated the stored procedure as well.